backend/sublet/views.py

```diff
@@ -0,0 +1,197 @@
+    serializer_class = SubletSerializer
+
+    def get_queryset(self):
+        # All Sublets for superusers
+        if self.request.user.is_superuser:
+            return Sublet.objects.all()
+
+        # All Sublets where expires_at hasn't passed yet for regular users
+        return Sublet.objects.filter(expires_at__gte=timezone.now())
+
+    def create(self, request, *args, **kwargs):
+        amenities = request.data.pop("amenities", [])
+
+        # check if valid amenities
+        try:
+            amenities = [Amenity.objects.get(name=amenity) for amenity in amenities]
+        except Amenity.DoesNotExist:
+            return Response({"amenities": "Invalid amenity"}, status=status.HTTP_400_BAD_REQUEST)
+
+        serializer = self.get_serializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        sublet = serializer.save()
+        sublet.amenities.set(amenities)
+        sublet.save()
+        return Response(serializer.data, status=status.HTTP_201_CREATED)
+
+    @action(detail=False, methods=["get"])
+    def browse(self, request):
+        """Returns a list of Sublets that match query parameters and user ownership."""
+        # Get query parameters from request (e.g., amenities, user_owned)
+        amenities = request.query_params.getlist("amenities")
+        subletter = request.query_params.get(
+            "subletter", False
+        )  # Defaults to False if not specified
+        starts_before = request.query_params.get("starts_before", None)
+        starts_after = request.query_params.get("starts_after", None)
+        ends_before = request.query_params.get("ends_before", None)
+        ends_after = request.query_params.get("ends_after", None)
+        min_price = request.query_params.get("min_price", None)
+        max_price = request.query_params.get("max_price", None)
+        beds = request.query_params.get("beds", None)
+        baths = request.query_params.get("baths", None)
+
+        queryset = Sublet.objects.all()
+
+        # Apply filters based on query parameters
+        if amenities:
+            queryset = queryset.filter(amenities__name__in=amenities)
+        if subletter.lower() == "true":
+            queryset = queryset.filter(subletter=request.user)
+        if starts_before:
+            queryset = queryset.filter(start_date__lt=starts_before)
+        if starts_after:
+            queryset = queryset.filter(start_date__gt=starts_after)
+        if ends_before:
+            queryset = queryset.filter(end_date__lt=ends_before)
+        if ends_after:
+            queryset = queryset.filter(end_date__gt=ends_after)
+        if min_price:
+            queryset = queryset.filter(min_price__gte=min_price)
+        if max_price:
+            queryset = queryset.filter(max_price__lte=max_price)
+        if beds:
+            queryset = queryset.filter(beds=beds)
+        if baths:
+            queryset = queryset.filter(baths=baths)
```

Comment:
Not sure if Justin/Tuneer would agree, but we could have the query params be literally the db query names so start_date__lt instead of starts_before, so we could literally just pass in **request.query_params

Reply:
don't think we need to worry about SQL injection attacks since Django will protect against that in the `filter` clause. but from the frontend side, it is probably nier for them to be declaring stuff as "max_price" rather than "max_price__lte" so this is fine imo.